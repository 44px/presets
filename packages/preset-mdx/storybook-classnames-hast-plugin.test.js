const path = require('path');
const fs = require('fs-extra');
const mdx = require('@mdx-js/mdx');
const prettier = require('prettier');
const plugin = require('./storybook-classnames-hast-plugin');

function format(code) {
  return prettier.format(code, {
    parser: 'babylon',
    printWidth: 100,
    tabWidth: 2,
    bracketSpacing: true,
    trailingComma: 'es5',
    singleQuote: true,
  });
}

describe('storybook-classnames-hast-plugin', () => {
  it('adds "storybook-mdx-XX" class name to every element', async () => {
    const content = await fs.readFile(path.resolve(__dirname, './test.mdx'), 'utf8');

    const result = mdx.sync(content, {
      hastPlugins: [plugin],
    });

    const code = format(result);

    expect(code).toMatchSnapshot();
  });
});
