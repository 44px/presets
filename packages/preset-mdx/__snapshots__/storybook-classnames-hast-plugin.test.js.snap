// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storybook-classnames-hast-plugin adds "storybook-mdx-XX" class name to every element 1`] = `
"export default ({ components, ...props }) => (
  <MDXTag name=\\"wrapper\\" components={components}>
    {/* storyKind = mdx syntax */}
    {/* storyName = heading */}
    <MDXTag
      name=\\"h1\\"
      components={components}
      props={{ className: 'storybook-mdx-h1' }}
    >{\`Heading 1\`}</MDXTag>
    <MDXTag
      name=\\"h2\\"
      components={components}
      props={{ className: 'storybook-mdx-h2' }}
    >{\`Heading 2\`}</MDXTag>
    <MDXTag
      name=\\"h3\\"
      components={components}
      props={{ className: 'storybook-mdx-h3' }}
    >{\`Heading 3\`}</MDXTag>
    <MDXTag
      name=\\"h4\\"
      components={components}
      props={{ className: 'storybook-mdx-h4' }}
    >{\`Heading 4\`}</MDXTag>
    <MDXTag
      name=\\"h5\\"
      components={components}
      props={{ className: 'storybook-mdx-h5' }}
    >{\`Heading 5\`}</MDXTag>
    <MDXTag
      name=\\"h6\\"
      components={components}
      props={{ className: 'storybook-mdx-h6' }}
    >{\`Heading 6\`}</MDXTag>
    <MDXTag name=\\"hr\\" components={components} props={{ className: 'storybook-mdx-hr' }} />
    <MDXTag name=\\"h1\\" components={components} props={{ className: 'storybook-mdx-h1' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h1\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Heading 1 With Link\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h2\\" components={components} props={{ className: 'storybook-mdx-h2' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h2\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Heading 2 With Link\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h3\\" components={components} props={{ className: 'storybook-mdx-h3' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h3\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Heading 3 With Link\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h4\\" components={components} props={{ className: 'storybook-mdx-h4' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h4\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Heading 4 With Link\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"hr\\" components={components} props={{ className: 'storybook-mdx-hr' }} />
    <MDXTag name=\\"h1\\" components={components} props={{ className: 'storybook-mdx-h1' }}>
      {\`Not link and \`}
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h1\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Link 1\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h2\\" components={components} props={{ className: 'storybook-mdx-h2' }}>
      {\`Not link and \`}
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h2\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Link 2\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h3\\" components={components} props={{ className: 'storybook-mdx-h3' }}>
      {\`Not link and \`}
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h3\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Link 3\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"h4\\" components={components} props={{ className: 'storybook-mdx-h4' }}>
      {\`Not link and \`}
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"h4\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Link 4\`}</MDXTag>
    </MDXTag>
    {/* storyName = lists */}
    <MDXTag name=\\"ul\\" components={components} props={{ className: 'storybook-mdx-ul' }}>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'storybook-mdx-li' }}
      >{\`List Item 1\`}</MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'storybook-mdx-li' }}
      >{\`List Item 2\`}</MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"em\\"
          components={components}
          parentName=\\"li\\"
          props={{ className: 'storybook-mdx-em' }}
        >{\`List Item 3 italic\`}</MDXTag>
      </MDXTag>
    </MDXTag>
    <MDXTag name=\\"hr\\" components={components} props={{ className: 'storybook-mdx-hr' }} />
    <MDXTag name=\\"ol\\" components={components} props={{ className: 'storybook-mdx-ol' }}>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ol\\"
        props={{ className: 'storybook-mdx-li' }}
      >{\`List Item 1\`}</MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ol\\"
        props={{ className: 'storybook-mdx-li' }}
      >{\`List Item 2\`}</MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ol\\"
        props={{ className: 'storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"a\\"
          components={components}
          parentName=\\"li\\"
          props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
        >{\`List Item 3 with link\`}</MDXTag>
      </MDXTag>
    </MDXTag>
    <MDXTag name=\\"hr\\" components={components} props={{ className: 'storybook-mdx-hr' }} />
    <MDXTag name=\\"ul\\" components={components} props={{ className: 'storybook-mdx-ul' }}>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'task-list-item storybook-mdx-task-list-item storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"input\\"
          components={components}
          parentName=\\"li\\"
          props={{
            type: 'checkbox',
            checked: true,
            disabled: true,
            className: 'storybook-mdx-input',
          }}
        />
        {\` \`}
        {\`Check Item 1\`}
      </MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'task-list-item storybook-mdx-task-list-item storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"input\\"
          components={components}
          parentName=\\"li\\"
          props={{
            type: 'checkbox',
            checked: false,
            disabled: true,
            className: 'storybook-mdx-input',
          }}
        />
        {\` \`}
        {\`Check Item 2\`}
      </MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'task-list-item storybook-mdx-task-list-item storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"input\\"
          components={components}
          parentName=\\"li\\"
          props={{
            type: 'checkbox',
            checked: true,
            disabled: true,
            className: 'storybook-mdx-input',
          }}
        />
        {\` \`}
        <MDXTag
          name=\\"strong\\"
          components={components}
          parentName=\\"li\\"
          props={{ className: 'storybook-mdx-strong' }}
        >{\`Check Item 3 bold\`}</MDXTag>
      </MDXTag>
      <MDXTag
        name=\\"li\\"
        components={components}
        parentName=\\"ul\\"
        props={{ className: 'task-list-item storybook-mdx-task-list-item storybook-mdx-li' }}
      >
        <MDXTag
          name=\\"input\\"
          components={components}
          parentName=\\"li\\"
          props={{
            type: 'checkbox',
            checked: false,
            disabled: true,
            className: 'storybook-mdx-input',
          }}
        />
        {\` \`}
        <MDXTag
          name=\\"a\\"
          components={components}
          parentName=\\"li\\"
          props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
        >{\`Check Item 4 with link\`}</MDXTag>
      </MDXTag>
    </MDXTag>
    {/* storyName = quotes */}
    <MDXTag
      name=\\"blockquote\\"
      components={components}
      props={{ className: 'storybook-mdx-blockquote' }}
    >
      <MDXTag
        name=\\"p\\"
        components={components}
        parentName=\\"blockquote\\"
        props={{ className: 'storybook-mdx-p' }}
      >{\`Note note note\`}</MDXTag>
      <MDXTag
        name=\\"p\\"
        components={components}
        parentName=\\"blockquote\\"
        props={{ className: 'storybook-mdx-p' }}
      >{\`Here here\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"hr\\" components={components} props={{ className: 'storybook-mdx-hr' }} />
    <MDXTag
      name=\\"blockquote\\"
      components={components}
      props={{ className: 'storybook-mdx-blockquote' }}
    >
      <MDXTag
        name=\\"p\\"
        components={components}
        parentName=\\"blockquote\\"
        props={{ className: 'storybook-mdx-p' }}
      >
        {\`Quotes with \`}
        <MDXTag
          name=\\"a\\"
          components={components}
          parentName=\\"p\\"
          props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
        >{\`link\`}</MDXTag>
      </MDXTag>
    </MDXTag>
    {/* storyName = text emphasizing */}
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"del\\"
        components={components}
        parentName=\\"p\\"
        props={{ className: 'storybook-mdx-del' }}
      >{\`Qwerty\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"em\\"
        components={components}
        parentName=\\"p\\"
        props={{ className: 'storybook-mdx-em' }}
      >{\`Qwerty\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"strong\\"
        components={components}
        parentName=\\"p\\"
        props={{ className: 'storybook-mdx-strong' }}
      >{\`Qwerty\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"inlineCode\\"
        components={components}
        parentName=\\"p\\"
        props={{ className: 'storybook-mdx-inlineCode' }}
      >{\`Inline code snippet\`}</MDXTag>
    </MDXTag>
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"p\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >
        <MDXTag
          name=\\"inlineCode\\"
          components={components}
          parentName=\\"a\\"
          props={{ className: 'storybook-mdx-inlineCode' }}
        >{\`Inline code with as a link\`}</MDXTag>
      </MDXTag>
    </MDXTag>
    {/* storyName = links */}
    <MDXTag name=\\"p\\" components={components} props={{ className: 'storybook-mdx-p' }}>
      <MDXTag
        name=\\"a\\"
        components={components}
        parentName=\\"p\\"
        props={{ href: 'https://storybook.js.org/', className: 'storybook-mdx-a' }}
      >{\`Storybook\`}</MDXTag>
    </MDXTag>
    {/* storyName = code */}
    <MDXTag
      name=\\"h3\\"
      components={components}
      props={{ className: 'storybook-mdx-h3' }}
    >{\`js\`}</MDXTag>
    <MDXTag name=\\"pre\\" components={components} props={{ className: 'storybook-mdx-pre' }}>
      <MDXTag
        name=\\"code\\"
        components={components}
        parentName=\\"pre\\"
        props={{ className: 'language-js storybook-mdx-language-js storybook-mdx-code' }}
      >{\`class Bar {
  constructor() {
  }
}

function foo() {
  return 'Bar';
}
\`}</MDXTag>
    </MDXTag>
    <MDXTag
      name=\\"h3\\"
      components={components}
      props={{ className: 'storybook-mdx-h3' }}
    >{\`ts\`}</MDXTag>
    <MDXTag name=\\"pre\\" components={components} props={{ className: 'storybook-mdx-pre' }}>
      <MDXTag
        name=\\"code\\"
        components={components}
        parentName=\\"pre\\"
        props={{ className: 'language-ts storybook-mdx-language-ts storybook-mdx-code' }}
      >{\`interface StorybookMdxOptions {
  showStoryCode?: boolean,
  includeStoryHeadings?: boolean,
  storyLang?: string,
  storybookApi?: 'annotations' | 'headings',
  defaultStoryKind?: DefaultStoryNameFn,
  defaultStoryName?: DefaultStoryNameFn | string,
}
\`}</MDXTag>
    </MDXTag>
    <MDXTag
      name=\\"h3\\"
      components={components}
      props={{ className: 'storybook-mdx-h3' }}
    >{\`css\`}</MDXTag>
    <MDXTag name=\\"pre\\" components={components} props={{ className: 'storybook-mdx-pre' }}>
      <MDXTag
        name=\\"code\\"
        components={components}
        parentName=\\"pre\\"
        props={{ className: 'language-css storybook-mdx-language-css storybook-mdx-code' }}
      >{\`.selector {
  color: 'red';
  padding: 10px;
}
\`}</MDXTag>
    </MDXTag>
    <MDXTag
      name=\\"h3\\"
      components={components}
      props={{ className: 'storybook-mdx-h3' }}
    >{\`jsx\`}</MDXTag>
    <button>This is JSX</button>
    {/* storyName = table */}
    <MDXTag
      name=\\"p\\"
      components={components}
      props={{ className: 'storybook-mdx-p' }}
    >{\`Default md table:\`}</MDXTag>
    <MDXTag name=\\"table\\" components={components} props={{ className: 'storybook-mdx-table' }}>
      <MDXTag
        name=\\"thead\\"
        components={components}
        parentName=\\"table\\"
        props={{ className: 'storybook-mdx-thead' }}
      >
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"thead\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"th\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-th' }}
          >{\`React\`}</MDXTag>
          <MDXTag
            name=\\"th\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-th' }}
          >{\`Vue\`}</MDXTag>
          <MDXTag
            name=\\"th\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-th' }}
          >{\`Angular\`}</MDXTag>
          <MDXTag
            name=\\"th\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-th' }}
          >{\`HTML\`}</MDXTag>
        </MDXTag>
      </MDXTag>
      <MDXTag
        name=\\"tbody\\"
        components={components}
        parentName=\\"table\\"
        props={{ className: 'storybook-mdx-tbody' }}
      >
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"tbody\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🚀\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🚀\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🚀\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🚀\`}</MDXTag>
        </MDXTag>
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"tbody\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🏅\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🏅\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🏅\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`🏅\`}</MDXTag>
        </MDXTag>
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"tbody\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`👍\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`👍\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`👍\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`👍\`}</MDXTag>
        </MDXTag>
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"tbody\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`💪\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`💪\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`💪\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`💪\`}</MDXTag>
        </MDXTag>
        <MDXTag
          name=\\"tr\\"
          components={components}
          parentName=\\"tbody\\"
          props={{ className: 'storybook-mdx-tr' }}
        >
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`😃\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`😃\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`😃\`}</MDXTag>
          <MDXTag
            name=\\"td\\"
            components={components}
            parentName=\\"tr\\"
            props={{ align: 'center', className: 'storybook-mdx-td' }}
          >{\`😃\`}</MDXTag>
        </MDXTag>
      </MDXTag>
    </MDXTag>
  </MDXTag>
);
"
`;
